packages/server/lib/watchers.js

```diff
@@ -60,27 +58,28 @@ class Watchers {
       },
     })
 
-    return _.each(files, (file) => {
+    return Promise.all(files.map((file) => {
       return this.watch(file, options)
-    })
+    }))
   }
 
   _add (filePath, watcher) {
-    this._remove(filePath)
-
-    this.watchers[filePath] = watcher
+    return this._remove(filePath)
+    .then(() => {
+      this.watchers[filePath] = watcher
+    })
   }
 
   _remove (filePath) {
     let watcher
 
     if (!(watcher = this.watchers[filePath])) {
-      return
+      return Promise.resolve()
     }
 
-    watcher.close()
-
-    return delete this.watchers[filePath]
+    return watcher.close().then(() => {
```

Comment:
Should we just delete first?

```js
delete this.watchers[filePath]
return watcher.close()
```

Or is there a specific reason for the ordering? I don't understand why we'd return the result of `delete`.

Reply:
It felt weird to delete a watcher and then call a `close()` function on it.

Reply:
Changed back in 8115c02